### DISCUSSION TITLE
Could you add a note to the editable installs documentation page that you can pass it build options?

### DISCUSSION THREAD

**Discussion by igo95862** [mesonbuild/meson-python#611] [Ideas]
I did not realize that you can pass it the build options the same way you can to other pip invocations. It took me some time to realize this. I was thinking the editable installs are special and don't take any build options.

For example, you can create C modules coverage:

```
./venv/bin/python -m pip install --no-build-isolation --config-settings=setup-args="-Db_coverage=true" --config-settings=editable-verbose=true --editable .
  
./venv/bin/python -m unittest
```

After that you can generate the LCOV html coverage report with `ninja -C build/cp311/ coverage-html ` and it works.

**Comment by Mohataseem89** [comment]
You can pass build options when doing an editable install but editable mode is not special in this regard `pip install -e .` still uses the same PEP 517 build backend interface, so anything you’d normally forward through `--config-settings` for a regular build will also be forwarded for editable builds

That means options like Meson setup arguments work the same way
```
python -m pip install \
  --no-build-isolation \
  --config-settings=setup-args="-Db_coverage=true" \
  --config-settings=editable-verbose=true \
  -e .
  ```

There is no separate mechanism for editable. the backend runs:
```
prepare_metadata_for_build_editable()
build_editable()
```
and pip simply forwards all `--config-settings` to meson-python, just like with wheels or sdists.
Your example for C-extension coverage is therefore expected:
```
ninja -C build/<tag>/ coverage-html
```

So the documentation could indeed mention explicitly that:
editable installs accept `--config-settings` just like normal builds, and
meson options can be propagated via `setup-args="-D…"`.

That might make the feature more discoverable for users who assume editable builds are “special” and do not expose backend options
